pymeasure/instruments/bkprecision/bkprecision8500.py

```diff
@@ -0,0 +1,283 @@
+PACKET_COMMAND_SET_CW_MODE_POWER = 0x2E
+PACKET_COMMAND_GET_CW_MODE_POWER = 0x2F
+PACKET_COMMAND_SET_CR_MODE_RESISTANCE = 0x30
+PACKET_COMMAND_GET_CR_MODE_RESISTANCE = 0x31
+PACKET_COMMAND_SET_REMOTE_SENSING_ENABLED = 0x56
+PACKET_COMMAND_GET_REMOTE_SENSING_ENABLED = 0x57
+PACKET_COMMAND_GET_INPUT_AND_STATE = 0x5F
+
+PACKET_STATUS_VALID = 0x80
+PACKET_STATUS_INCORRECT_CHEKSUM = 0x90
+PACKET_STATUS_PARAMETER_INCORRECT = 0xA0
+PACKET_STATUS_UNRECOGNIZED_COMMAND = 0xB0
+PACKET_STATUS_INVALID_COMMAND = 0xC0
+
+
+def calculate_checksum(packet):
+    return sum(packet[0:PACKET_LENGTH - 1]) % 256
+
+
+def make_packet(address, command, command_data):
+    packet = bytearray(PACKET_LENGTH)
+    packet[PACKET_HEADER_INDEX] = PACKET_HEADER
+    packet[PACKET_ADDRESS_INDEX] = address
+    packet[PACKET_COMMAND_INDEX] = command
+    packet[PACKET_COMMAND_DATA_START_INDEX:PACKET_COMMAND_DATA_START_INDEX + len(command_data)] = command_data
+    packet[PACKET_CHECKSUM_INDEX] = calculate_checksum(packet)
+    return packet
+
+
+def value_to_little_endian(value):
+    return [value & 0xff,
+            value >> 8 & 0xff,
+            value >> 16 & 0xff,
+            value >> 24 & 0xff]
+
+
+def little_endian_to_value(arr):
+    return arr[0] + arr[1] * 256 + arr[2] * 256 * 256 + arr[3] * 256 * 256 * 256
+
+
+def make_command_data(size, scale, value):
+    if size > 1:
+        return value_to_little_endian(int(scale * value))
+    else:
+        return [int(scale * value)]
+
+
+def ensure_valid_checksum(packet):
+    if (calculate_checksum(packet) != packet[PACKET_CHECKSUM_INDEX]):
+        raise Exception('Received packet with incorrect checksum')
+    return packet
+
+
+def ensure_valid_status(packet):
+    assert packet[PACKET_COMMAND_INDEX] == PACKET_COMMAND_STATUS
+    ensure_valid_checksum(packet)
+    status = packet[PACKET_COMMAND_DATA_START_INDEX]
+    if (status != PACKET_STATUS_VALID):
+        status_map = {PACKET_STATUS_INCORRECT_CHEKSUM: "Checksum incorrect",
+                      PACKET_STATUS_PARAMETER_INCORRECT: "Parameter incorrect",
+                      PACKET_STATUS_UNRECOGNIZED_COMMAND: "Unrecognized command",
+                      PACKET_STATUS_INVALID_COMMAND: "Invalid command"}
+        raise Exception('Unexpected status: ' + status_map.get(status, "Unknown status"))
+
+
+def make_get_str(packet_command, scale=1, size=1, offset=0):
+    return "{:d} {:d} {:d} {:d}".format(packet_command, size, scale, offset)
+
+
+def make_set_str(packet_command, specifier="%d", scale=1, size=1):
+    return "{:d} {:s} {:d} {:d}".format(packet_command, specifier, size, scale)
+
+
+class BKPrecision8500(Instrument):
+    """ Represents the BK Precision 8500 Programmable DC Electronic Load and
+    provides a high-level interface for interacting with the instrument.
+    """
+
+    def __init__(self, resourceName, baud_rate=4800, **kwargs):
+        super().__init__(
+            resourceName,
+            "BK Precision 8500",
+            **kwargs
```

Comment:
Being a non SCPI instrument, you need to add `includeSCPI=False`, in parameters.
I would also try to add `write_termination=""` and `read_termination=""`. This could be sufficient to handle binary values. Also you may want to check `binary_values` and `write_binary_values` in the `VISAAdapter` and `SerialAdapter` classes.